app/src/main/java/com/example/android/uamp/fragments/NowPlayingScreen.kt

```diff
@@ -0,0 +1,220 @@
+    currentMediaItem?.let { mediaItem ->
+        NowPlayingDescription(
+            mediaItem,
+            nowPlayingFragmentViewModel,
+            mainActivityViewModel,
+            navController
+        )
+    }
+}
+
+/**
+ * This particular instance of NowPlayingDescription serves to showcase the current media item being played
+ * on the screen
+ *
+ * @param mediaItem current mediaItem being played
+ * @param nowPlayingFragmentViewModel to reference NowPlayingFragmentViewModel functions
+ * @param mainActivityViewModel to reference MainActivityViewModel functions
+ * @param navController to navigate to SettingsScreen
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun NowPlayingDescription(
+    mediaItem: MediaItem,
+    nowPlayingFragmentViewModel: NowPlayingFragmentViewModel,
+    mainActivityViewModel: MainActivityViewModel,
+    navController: NavController
+) {
+    if (mediaItem.equals(null)) {
+        Text(text = "Media Item is null")
+    } else {
+        Surface() {
+            NowPlaying(mediaItem, nowPlayingFragmentViewModel, mainActivityViewModel, navController)
+        }
+    }
+}
+
+/**
+ * NowPlaying describes the UI of the NowPlayingScreen which recomposes upon changes in the state of
+ * currentMediaItem's position, duration, and change in playback (play/pause).
+ *
+ * @param mediaItem current mediaItem being played
+ * @param nowPlayingFragmentViewModel to reference NowPlayingFragmentViewModel functions
+ * @param mainActivityViewModel to reference MainActivityViewModel functions
+ * @param navController to navigate to SettingsScreen
+ */
+@SuppressLint("PrivateResource")
+@Composable
+private fun NowPlaying(
+    mediaItem: MediaItem,
+    nowPlayingFragmentViewModel: NowPlayingFragmentViewModel,
+    mainActivityViewModel: MainActivityViewModel,
+    navController: NavController
+) {
+    val position: Long by nowPlayingFragmentViewModel.mediaPosition.observeAsState(0)
+    val duration: Long by nowPlayingFragmentViewModel.mediaDuration.observeAsState(0)
+    val buttonRes: Int? by nowPlayingFragmentViewModel.mediaButtonRes.observeAsState()
+
+    Log.d("DURATION", "${duration}")
+    Log.d("POSITION", "${position}")
+
+    val positionMinute = (position.div(1000)).div(60)
+    val positionSecond = "%02d".format((position.div(1000)).mod(60))
+
+    val durationMinute = (duration.div(1000)).div(60)
+    val durationSecond = "%02d".format((duration.div(1000)).mod(60))
+
+    val buttonWidth = dimensionResource(id = R.dimen.exo_media_button_width)
+    val margins = dimensionResource(id = R.dimen.text_margin)
+
+    val mediaItemArtwork = mediaItem.mediaMetadata.artworkUri
+
+    Column(modifier = Modifier.fillMaxSize()) {
+        Row(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalArrangement = Arrangement.End
+        ) {
+            TopAppBar(
+                title = { Text("") },
+                navigationIcon = {
```

Comment:
nit: use `actions` here instead of `navigationIcon` so that the settings gear appears on the right side, which is where I'd normally expect it to appear.